karton/system/system.py

```diff
@@ -107,32 +121,118 @@ def gc_collect_tasks(self) -> None:
                     task.headers.get("receiver", "<unknown>"),
                 )
             if will_delete:
-                self.backend.delete_task(task)
-                self.backend.increment_metrics(
-                    KartonMetrics.TASK_GARBAGE_COLLECTED,
-                    task.headers.get("receiver", "unknown"),
-                )
+                to_delete.append(task)
```

Comment:
(small possible refactoring: remove `will_delete = True` lines and using `to_delete.append(task)` instead)